### ISSUE TITLE
State is not updating POW (R2)

### ISSUE THREAD

**Issue by phreaq** [peterbuga/HASS-sonoff-ewelink#128]
Hi folks,

I'm trying out this component and I'm able to see my Sonoff POW as device, and it is reading it's initial values for Amps, Volts, Watts, etc. however... if the state changes (e.g., the load increases), the status is not updating within Hassio on the States page.

I have the following in my configuration.yaml

> sonoff:
>   username: me@somemail.com
>   password: Illnevertell
>   api_region: 'us'
>   grace_period: 60
>   scan_interval: 60

any thoughts?

**Comment by peterbuga** [comment]
please see this post https://community.home-assistant.io/t/sonoff-ewelink-component-for-original-firmware/81369/589?u=peterbuga and the responses following them and let me know if anything matches.

on what firmware version is the Sonoff Pow (R2 i assume) ?

**Comment by phreaq** [comment]
thanks for the reply, 

yes, that post describes what I am seeing, and yes, I am using POW R2 using 3.3.

I'm a bit confused with the final resolution.

I do have two accounts, one for HA, and one for my mobile access. 

I first created/added the POW R2 device using the HA account and shared it to my mobile account.

Do I maintain that setup? 

Do I use LAN mode? 

Do I keep logged into the mobile account (on my mobile), or do I swap things around, running the 'original' account on my mobile, and the shared to account on HA?

**Comment by peterbuga** [comment]
unfortunately there's no final resolution ... (yet, if any). i was just curious if you're experiencing the same problem(s) in order to evaluate if it's a recurring problem.

**Comment by neomvll** [comment]
Same problem with my 2 POW R2. The sonoff dev know this issue, but have no date to solve. Really bad!
https://support.itead.cc/support/discussions/topics/11000027580/page/2?url_locale=

"Hi, We will solve this as soon as possible. So sorry for this inconvenience"

**Comment by neomvll** [comment]
Hi guys, 3.3.1 available.

**Comment by neomvll** [comment]
shit, after 15 hours, stop working again.

**Comment by peterbuga** [comment]
@neomvll can you please give a bit more details, personally I don't understand if 3.3.1 fixed anything in the first place or not 😅

**Comment by neomvll** [comment]
> @neomvll can you please give a bit more details, personally I don't understand if 3.3.1 fixed anything in the first place or not 😅

Well, yesterday, after update, POW R2 back to normal, sending updated values about power comsume, but today nothing works again. I think the update not solved the issue.

**Comment by ilgeko** [comment]
The same for me. Everything is updated. I have a dedicated ewelink account for homeassistant. The states updates only if i have the app open, active and not in lan mode.

**Comment by neomvll** [comment]
> The same for me. Everything is updated. I have a dedicated ewelink account for homeassistant. The states updates only if i have the app open, active and not in lan mode.

Yes, it's pretty bugged
We can complaing here too:
https://twitter.com/ITeadstudio/status/1171756272812617728
and
https://support.itead.cc/support/discussions/topics/11000027580

**Comment by elserge82** [comment]
It updates ever minute in case the Ewelink app is open in my case.

**Comment by peterbuga** [comment]
@elserge82 yes, it has been reported before a similar _positive output_ (if we can call it like this) but unfortunately i'm not risking to update my POW R2 to debug it & possibly to lose the states-update functionality (i don't have a serial2usb adapter to go back and forth firmware versions 😞)

**Comment by naamah75** [comment]
Same issue here with the sonoff last release 3.3.0.
Home Assistant can't update the status.
The only workaround that I've found is to left the app opened on the smartphone.

**Comment by peterbuga** [comment]
@naamah75 or anyone else could provide me an [debug](https://github.com/peterbuga/HASS-sonoff-ewelink#debug-log-generation--new-device--new-features-requests) output of POW (R2) with the app open? 

basic steps: share **only** POW/R2 to a 2nd (3rd) account, add it to HA, follow the above steps to enable debug-mode, turn on/off the POW device from eWeLink app (that has the original account) and/or plug something to the sonoff device to make it send back some current/amps values back

post the output here or put in a file if it's too big and link it
thanks.

**Comment by alesoft73** [comment]
Hi, i have the same problem, work a few minute, after unknown.
MAny thanks

**Comment by naamah75** [comment]
@peterbuga below my logs...
`--------------COPY-FROM-HERE--------------

2019-11-12 16:03:05.453 [D] {"error": 406, "msg": "{\"oauth_authorise\":\"tokenInfo is not exit:a9200d3ca215dc47c41d5b8f050d72248a9fb234\"}"}

2019-11-12 16:03:17.189 [D] {"error": 406, "msg": "{\"oauth_authorise\":\"tokenInfo is not exit:a9200d3ca215dc47c41d5b8f050d72248a9fb234\"}"}

2019-11-12 16:03:26.530 [D] {"error": 406, "msg": "{\"oauth_authorise\":\"tokenInfo is not exit:a9200d3ca215dc47c41d5b8f050d72248a9fb234\"}"}

2019-11-12 16:03:38.050 [D] {"error": 406, "msg": "{\"oauth_authorise\":\"tokenInfo is not exit:a9200d3ca215dc47c41d5b8f050d72248a9fb234\"}"}

---------------END-OF-COPY----------------`

**Comment by peterbuga** [comment]
@naamah75 not good, you didn't use a 2nd ewelink account :(

**Comment by naamah75** [comment]
Uhm... ![bitmoji](https://render.bitstrips.com/v2/cpanel/d970d2f4-7af2-437c-b85f-469dfe2f5079-93c9e731-722e-413a-b2b3-4160f18049d2-v1.png?transparent=1&palette=1&width=246)
What you mean for "2nd" account?

**Comment by naamah75** [comment]
@peterbuga ah... ok... now I get! I will try it later!

**Comment by naamah75** [comment]
@peterbuga well, now I must got the right sequence...
`--------------COPY-FROM-HERE--------------

2019-11-12 18:22:35.594 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "1000642b9e", "params": {"uiActive": 60}, "sequence": "1573579358081", "tempRec": "1000642b9e", "ts": 0, "userAgent": "app"}

2019-11-12 18:22:35.718 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9825", "params": {"uiActive": 60}, "sequence": "1573579358241", "tempRec": "10007c9825", "ts": 0, "userAgent": "app"}

2019-11-12 18:22:35.743 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9834", "params": {"uiActive": 60}, "sequence": "1573579358329", "tempRec": "10007c9834", "ts": 0, "userAgent": "app"}

2019-11-12 18:22:35.748 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9829", "params": {"uiActive": 60}, "sequence": "1573579358304", "tempRec": "10007c9829", "ts": 0, "userAgent": "app"}

2019-11-12 18:22:35.829 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9474", "params": {"uiActive": 60}, "sequence": "1573579358296", "tempRec": "10007c9474", "ts": 0, "userAgent": "app"}

2019-11-12 18:22:36.132 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "1000642b9e", "from": "device", "params": {"current": "0.05", "power": "6.43", "voltage": "229.91"}, "ts": 0, "userAgent": "device"}

2019-11-12 18:22:36.379 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "1000642b9e", "params": {"uiActive": 60}, "sequence": "1573579359059", "tempRec": "1000642b9e", "ts": 0, "userAgent": "app"}

2019-11-12 18:22:36.411 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "1000642b9e", "params": {"uiActive": 60}, "sequence": "1573579359096", "tempRec": "1000642b9e", "ts": 0, "userAgent": "app"}

2019-11-12 18:22:36.557 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "1000642b9e", "params": {"uiActive": 60}, "sequence": "1573579359107", "tempRec": "1000642b9e", "ts": 0, "userAgent": "app"}

2019-11-12 18:22:37.096 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9829", "from": "device", "params": {"current": "0.00", "power": "0.00", "voltage": "229.02"}, "ts": 0, "userAgent": "device"}

2019-11-12 18:22:37.285 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9834", "from": "device", "params": {"current": "0.74", "power": "96.02", "voltage": "226.32"}, "ts": 0, "userAgent": "device"}

2019-11-12 18:22:37.564 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9474", "from": "device", "params": {"current": "0.07", "power": "2.73", "voltage": "228.05"}, "ts": 0, "userAgent": "device"}

2019-11-12 18:22:37.680 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9825", "from": "device", "params": {"current": "0.00", "power": "0.00", "voltage": "227.76"}, "ts": 0, "userAgent": "device"}

2019-11-12 18:22:38.020 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9834", "params": {"uiActive": 60}, "sequence": "1573579360186", "tempRec": "10007c9834", "ts": 0, "userAgent": "app"}

2019-11-12 18:22:38.038 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9825", "params": {"uiActive": 60}, "sequence": "1573579360694", "tempRec": "10007c9825", "ts": 0, "userAgent": "app"}

2019-11-12 18:22:38.047 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9474", "params": {"uiActive": 60}, "sequence": "1573579360627", "tempRec": "10007c9474", "ts": 0, "userAgent": "app"}

2019-11-12 18:22:38.058 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9829", "params": {"uiActive": 60}, "sequence": "1573579360021", "tempRec": "10007c9829", "ts": 0, "userAgent": "app"}

2019-11-12 18:22:38.106 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9834", "params": {"uiActive": 60}, "sequence": "1573579360194", "tempRec": "10007c9834", "ts": 0, "userAgent": "app"}

2019-11-12 18:22:38.132 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "1000642b9e", "from": "device", "params": {"current": "0.05", "power": "6.40", "voltage": "229.91"}, "ts": 0, "userAgent": "device"}

2019-11-12 18:22:39.096 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9829", "from": "device", "params": {"current": "0.00", "power": "0.00", "voltage": "228.59"}, "ts": 0, "userAgent": "device"}

2019-11-12 18:22:39.308 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9834", "from": "device", "params": {"current": "0.74", "power": "95.98", "voltage": "226.60"}, "ts": 0, "userAgent": "device"}

2019-11-12 18:22:39.560 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9474", "from": "device", "params": {"current": "0.07", "power": "2.73", "voltage": "228.05"}, "ts": 0, "userAgent": "device"}

2019-11-12 18:22:39.693 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9825", "from": "device", "params": {"current": "0.00", "power": "0.00", "voltage": "228.04"}, "ts": 0, "userAgent": "device"}

2019-11-12 18:22:41.769 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9825", "from": "device", "params": {"current": "0.08", "power": "1.75", "voltage": "228.33"}, "ts": 0, "userAgent": "device"}

2019-11-12 18:22:43.728 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9825", "from": "device", "params": {"current": "0.07", "power": "1.75", "voltage": "228.33"}, "ts": 0, "userAgent": "device"}

2019-11-12 18:22:45.940 [s] {"entity_id": "switch.sonoff_10007c9829", "outlet": null, "state": "off"}

2019-11-12 18:22:45.940 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9829", "from": "device", "params": {"switch": "off"}, "seq": "193", "sequence": "1573579368232", "ts": 0, "userAgent": "device"}

2019-11-12 18:22:47.086 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9829", "from": "device", "params": {"current": "0.00", "power": "0.00", "voltage": "229.13"}, "ts": 0, "userAgent": "device"}

2019-11-12 18:22:47.292 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9834", "from": "device", "params": {"current": "0.60", "power": "95.61", "voltage": "227.44"}, "ts": 0, "userAgent": "device"}

2019-11-12 18:22:49.285 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9834", "from": "device", "params": {"current": "0.07", "power": "95.61", "voltage": "226.88"}, "ts": 0, "userAgent": "device"}

2019-11-12 18:22:51.284 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9834", "from": "device", "params": {"current": "0.08", "power": "71.71", "voltage": "226.46"}, "ts": 0, "userAgent": "device"}

2019-11-12 18:22:53.289 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9834", "from": "device", "params": {"current": "0.00", "power": "0.00", "voltage": "226.32"}, "ts": 0, "userAgent": "device"}

2019-11-12 18:22:55.686 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9825", "from": "device", "params": {"current": "0.06", "power": "1.99", "voltage": "227.47"}, "ts": 0, "userAgent": "device"}

2019-11-12 18:22:57.685 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9825", "from": "device", "params": {"current": "0.00", "power": "0.00", "voltage": "227.19"}, "ts": 0, "userAgent": "device"}

2019-11-12 18:23:00.057 [s] {"entity_id": "switch.sonoff_10007c9829", "outlet": null, "state": "on"}

2019-11-12 18:23:00.057 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9829", "from": "device", "params": {"switch": "on"}, "seq": "195", "sequence": "1573579381790", "ts": 0, "userAgent": "device"}

2019-11-12 18:23:01.080 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9829", "from": "device", "params": {"current": "0.00", "power": "0.00", "voltage": "229.31"}, "ts": 0, "userAgent": "device"}

2019-11-12 18:23:01.563 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "10007c9474", "from": "device", "params": {"current": "0.07", "power": "2.68", "voltage": "228.33"}, "ts": 0, "userAgent": "device"}

---------------END-OF-COPY----------------`

**Comment by peterbuga** [comment]
@naamah75 perfect you got it right this time 👍
now there’s a _good news bad news_ kind of moment: it looks like the fix might not be as complicated as i might’ve imagined, the not so nice part is that i really need to rethink the whole component to accomodate this change 😔

this thing and coupled with the fact that the more devices i add the more spaghetti code i create and it’s just not manageable

long story short it’ll take a while till i’ll push this change, thanks for the debug output! 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by naamah75** [comment]
@peterbuga no problem! we'll consider that the good news is most imporant that the bad news! in any case really thanks for your work! 👍👍👍

**Comment by alesoft73** [comment]
Many thanks for your support!!!!!! @peterbuga 
I send you a supprt for a coffee ;)

**Comment by peterbuga** [comment]
@alesoft73 ☕️ received ! thank you 🙌!

**Comment by naamah75** [comment]
I didn't fine the "buy me a coffe" link for @peterbuga ... where I can find it?

**Comment by alesoft73** [comment]
> 
> 
> I didn't fine the "buy me a coffe" link for @peterbuga ... where I can find it?

https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=WXDJJFCULBVSL&source=url

**Comment by naamah75** [comment]
... a coffee ☕️ for the bothering moments during the hard coding is arriving! 👍

**Comment by peterbuga** [comment]
@naamah75 thanks a lot to you & everyone who helps in any way they can 😊!

**Comment by ashoktvm** [comment]
Cant wait for an update. Thanks for all the work you put into this

**Comment by nick2525** [comment]
Second account will fix bug or not?

**Comment by peterbuga** [comment]
no, it needs a complete logic re-do
